build-info-extractor-gradle/src/main/groovy/org/jfrog/gradle/plugin/artifactory/task/ExtractModuleTask.java

```diff
@@ -12,7 +12,7 @@
 import java.lang.reflect.InvocationTargetException;
 
 public class ExtractModuleTask extends DefaultTask {
-    @OutputFile
+
     public RegularFileProperty moduleFile;
```

Comment:
Can you make the field `private` and `final`? Or would that be a breaking change?

Reply:
I changed it to `private`. However, changing it to final would require making a change in the constructor:

```
   public ExtractModuleTask() {
        try {
            this.moduleFile = getProject().getObjects().fileProperty();
        } catch (NoSuchMethodError e) {
            // Gradle 4.x
            this.moduleFile = invokeMethod(getProject().getLayout(), "fileProperty");
        }
    }
```

Since `moduleFile` could potentially be assigned twice so java compiler doesn't like it

This leaves me again in https://github.com/jfrog/build-info/issues/302 where a better story around testability of this plugin would help to make changes with more confident. I think having it as `private` to start is a move in the right direction

Reply:
Actually, I think this is better: https://github.com/jfrog/build-info/pull/326/commits/0b8a578296eca1cdaaff61830029b59ddc880e42  instead of relying on the try/catch